cmd/dev_client/main.go

```diff
@@ -120,8 +120,12 @@ func main() {
 		AuthHandler: authHandler,
 		KeyFolder:   keyFolder,
 		Client:      &http.Client{Transport: &http.Transport{TLSClientConfig: tlsConfig}},
-		Version:     client.GetVersion(),
 	}
 
-	client.Run(cli, &client.VisibilityParams{Logf: log.Printf, Errorf: log.Printf, Metrics: func(map[string]uint64) {}}, tokenEndpoint, clientID)
+	client.Run(
+		cli,
+		&client.VisibilityParams{log.Printf, log.Printf, func(map[string]uint64) {}},
+		tokenEndpoint,
+		clientID,
+		"")
```

Comment:
looks like the new parameter is never used -- can you add it as a flag and consume it here?

seems like someone would have to fork + recompile from source to leverage this functionality.

Reply:
With the default value of the flag being the hard-coded const?

Reply:
that would be reasonable to me

Reply:
As discussed offline, this will be a substantial amount of work.

- None of the commands (or even the Run function) parse the User’s token
- This work is done by the `authHandler` function. The `authHandler` function in our case parses the token file from a well known location. It’s actually not even included in the Github code, our `authHandler` function lies in our internal codebase
- To get the `authHandler` function information about how to parse the token, we would need to change the interface contract. Right now the `authHandler` function takes no parameters and just returns a string (the access token).
- For us to support a configurable token file location, we’d need to pass all this information down to the `authHandler` function

A large part of the problem stems from the fact that there isn’t really a good configuration story for Knox (since there is no configuration file). Unless we add a configuration layer, making changes like the ones you propose are difficult to do